data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/BulkOperationWrapper.java

```diff
@@ -76,6 +73,7 @@ public EventHandle getEventHandle() {
 
     public void releaseEventHandle(boolean result) {
         if (eventHandle != null) {
+            sink.updateLatencyMetrics(eventHandle);
```

Comment:
Ultimately we want this called because `release` is called. Instead of doubling up calls, how about we configure the `EventHandle` to provide this type of callback?

```
interface EventHandle {
  ...
  void onRelease(BiConsumer<EventHandle, Boolean> reseaseConsumer);
  ...
}
```

Then, in `OpenSearchSink`, be sure to make this call.

```
event.getHandle().onRelease((handle, result) -> updateLatencyMetrics(handle));
```

The current design results in too many classes having to be aware of these interactions.